cmd/collector/app/processor/processor.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package processor
+
+import (
+	"io"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+var (
+	_ Batch = (*SpansV1)(nil)
+	_ Batch = (*SpansV2)(nil)
+)
+
+// Batch is a batch of spans passed to the processor.
+type Batch interface {
+	// GetSpans delegates to the appropriate function based on the data model version.
+	GetSpans(v1 func(spans []*model.Span), v2 func(traces ptrace.Traces))
```

Comment:
is `GetSpans` the correct name here? This function doesn't return anything and rather just runs one of the provided functions

Reply:
It does provide access (Get) to the underlying data, just by using a visitor pattern instead of return value. We could call it Visit() but I think it's more consistent to call Get as all the other functions of the interface.